draft-ietf-quic-http.md

```diff
@@ -761,8 +762,8 @@ This results in sending a QUIC CONNECTION_CLOSE frame to the peer; the error
 code in this frame indicates to the peer why the connection is being closed.
 See {{errors}} for error codes which can be used when closing a connection.
 
-Before closing the connection, a GOAWAY MAY be sent to allow the client to retry
-some requests.  Including the GOAWAY frame in the same packet as the QUIC
+Before closing the connection, a GOAWAY frame MAY be sent to allow the client to
+retry some requests.  Including the GOAWAY frame in the same packet as the QUIC
 CONNECTION_CLOSE frame improves the chances of the frame being received by
 clients.
```

Comment:
With the parallel changes and discussion around GOAWAY I'm really struggling to understand the suggestion of `including the GOAWAY frame in the same packet as the QUIC CONNECTION_CLOSE`

Reply:
I think the text is suggesting that if they are in different packets and the CONNECTION_CLOSE arrives first, then the GOAWAY might be ignored?

Reply:
The intent is: If a client receives a GOAWAY first, it knows what it can retry, hence the value of putting both frames in a single packet.

Reply:
We're mixing together processing of a QUIC frame and an application-layer frame delivered inside of STREAM frames. I'm not sure trying to make assertions on the ordering of those things is the best way to do it. My I should open this out as a separate issue.

Reply:
What if they are in the same packet but the STREAM frame comes after the CONNECTION_CLOSE?

Reply:
Then likely the CONNECTION_CLOSE will be processed before the GOAWAY and the receiver will process the CONNECTION_CLOSE and ignore the GOAWAY.  I think that's how this has to work, but maybe I'm wrong?

Reply:
I think you're right. Thinking some more, I think the phrasing could be read to say that inclusion improves the probability of delivery, which is odd. When actually what it is trying to articulate is that processing order has some importance. However, just because the data arrives at the same point in time doesn't mean much IMO, that GOAWAY frame could just as well be blocked by other stuff waiting to be restransmitted on the stream.

Reply:
Admittedly, it's best effort.  This also implies that there might be value in permitting the application layer to drain whatever data has arrived, even if a CONNECTION_CLOSE has also been received.